httpcore5/src/main/java/org/apache/hc/core5/reactor/SingleCoreIOReactor.java

```diff
@@ -51,8 +53,14 @@
 import org.apache.hc.core5.util.Args;
 import org.apache.hc.core5.util.Timeout;
 
-class SingleCoreIOReactor extends AbstractSingleCoreIOReactor implements ConnectionInitiator {
+import static org.apache.hc.core5.util.ReflectionUtils.getExtendedSocketOptionOrNull;
 
+@Internal
+public class SingleCoreIOReactor extends AbstractSingleCoreIOReactor implements ConnectionInitiator {
+
+    public static final String TCP_KEEPIDLE = "TCP_KEEPIDLE";
```

Comment:
@kkewwei This is not the right place. Please move those constants to a separate file and mark it internal. The `org.apache.hc.core5.io` might be a good place.